hs_core/views/utils.py

```diff
@@ -57,18 +58,22 @@ def authorize(request, res_id, edit=False, view=False, full=False, superuser=Fal
     except ObjectDoesNotExist:
         raise NotFound(detail="No resource was found for resource id:%s" % res_id)
 
-    if user.is_authenticated():
+    if discoverable and (edit is False and view is False and full is False and superuser is False):
+        authorized = res.raccess.discoverable
+    elif user.is_authenticated():
         has_edit = user.uaccess.can_change_resource(res)
         has_view = user.uaccess.can_view_resource(res)
```

Comment:
@hyi As a matter a principle, I think we should have one and only one place where authorization is determined (the authorization function).  In this case, one could do:

    if authorize(discoverable=True):
        if authorize(view=True):
            # Look at content files ...

Would this achieve the use case you had in mind?